src/remote_action_middleware.js

```diff
@@ -1,6 +1,15 @@
+import constants from '../src/constants';
+
 export default (socket) => (store) => (next) => (action) => {
   if (action.meta && action.meta.remote) {
     socket.emit(action.type, action);
   }
+
+  if(action.onSuccess) {
+    const { type, onSuccess } = action.onSuccess;
+
+    socket.on(type, onSuccess);
```

Comment:
It's to catch the server emitting an on success event, so like:
1. Client emits add order event, and creates on event success;
2. Server receives action and on completion, emits on success event that the client catches and executes the callback.